packages/merge-styles/src/concatStyleSetsWithProps.ts

```diff
@@ -15,7 +15,11 @@ export function concatStyleSetsWithProps<TStyleProps, TStyleSet extends IStyleSe
   const result: Array<DeepPartial<TStyleSet>> = [];
   for (const styles of allStyles) {
     if (styles) {
-      result.push(typeof styles === 'function' ? styles(styleProps) : styles);
+      if (typeof styles === 'function') {
+        result.push((styles as IStyleFunction<TStyleProps, TStyleSet>)(styleProps));
```

Comment:
we can add these assertions within original ternary = keeping the bundle size on same level